db/db_impl_open.cc

```diff
@@ -99,7 +99,9 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src) {
     result.db_paths.emplace_back(dbname, std::numeric_limits<uint64_t>::max());
   }
 
-  if (result.use_direct_reads && result.compaction_readahead_size == 0) {
```

Comment:
I omit here because I grep all the `use_direct_writes` and this place didn't show up...